umm-lib/test/cmr/umm/test/echo10/granule.clj

```diff
@@ -38,7 +45,8 @@
     (let [xml (echo10/umm->echo10-xml granule)
           parsed (g/parse-granule xml)
           expected-parsed (umm->expected-parsed-echo10 granule)]
-      (= parsed expected-parsed))))
+      ;; remove crid-ids and feature-ids because they are not supported in echo10. 
+      (= (remove-crid-ids-and-feature-ids parsed) expected-parsed))))
```

Comment:
Ideally we shouldn't be modifying the parsed value. Why can't we do everything in expected-parsed?

Reply:
I found the old PR with comments. It looks like we aren't modifying expected-parsed correctly. Instead of using `dissoc` to remove keys we should be setting nil values for those keys. I'll add suggestions to the code.

Reply:
I know exactly what you meant, I tried that, but then there is the case when data-granule being nil, then I will have this extra :crid-ids nil :feature-ids nil in the expected... Let me take another look to see if I could include that logic in the expected.